tests/tests-hz38/src/main/java/com/hazelcast/simulator/tests/quorum/QuorumMapTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.simulator.tests.quorum;
+
+import static org.junit.Assert.assertEquals;
+
+import com.hazelcast.core.IMap;
+import com.hazelcast.quorum.QuorumException;
+import com.hazelcast.simulator.test.AbstractTest;
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Teardown;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+
+public class QuorumMapTest extends AbstractTest {
+
+	// properties
+	private IMap<Long, Long> map;
+	public int keyCount = 100000;
+
+	private volatile boolean flag;
+
+
+	@Setup
+	@SuppressWarnings("unchecked")
+	public void setup() {
+		map = targetInstance.getMap(name);
+
+	}
+
+	@TimeStep
+	public void put(BaseThreadState state) {
+		boolean testStatus = true;
+		long key = state.randomInt(keyCount);
+		try {
+			map.put(key, 0L);
+		} catch (QuorumException qe) {
+			testStatus = false;
```

Comment:
You could use `fail("QuorumException on map.put(" + key + "): " + qe.getMessage())` here instead of that boolean variable and `assertEquals()`. The thrown exception should be the same (`AssertionError`), but this would include some more information.